crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -37,6 +38,79 @@ impl DynamicScene {
         Self::from_world(&scene.world, type_registry)
     }
 
+    /// Create a Dynamic Scene with specific entities.
+    ///
+    /// The generic parameter is used as a [`Query`] filter.
+    ///
+    /// The created scene will include only the entities that match the query
+    /// filter provided. All components that impl `Reflect` will be included.
+    ///
+    /// # Example
+    ///
+    /// Here, the function is creating a scene with all the componentA's in the world, but none of the componentB's.
+    ///
+    /// ```
+    /// use bevy_ecs::prelude::*;
+    /// use bevy_scene::DynamicScene;
+    /// use bevy_app::AppTypeRegistry;
+    ///
+    /// #[derive(Component)]
+    /// struct ComponentA;
+    ///
+    /// #[derive(Component)]
+    /// struct ComponentB;
+    ///
+    /// let mut world = World::new();
+    /// world.init_resource::<AppTypeRegistry>();
+    ///
+    /// let my_scene = DynamicScene::from_query_filter::<(
+    ///   With<ComponentA>,
+    ///   Without<ComponentB>,
+    /// )>(&mut world);
+    /// ```
+    pub fn from_query_filter<F>(world: &mut World) -> Self
+    where
+        F: ReadOnlyWorldQuery + 'static,
+    {
+        let mut query = world.query_filtered::<Entity, F>();
+
+        let type_registry = world
+            .get_resource::<AppTypeRegistry>()
```

Comment:
IMO we should probably force users to pass in this type explicitly, in order to clarify which type registry they want to use.

Reply:
Wouldn't there just be one type registry per world? Or per app? How could they be multiple?

Reply:
Ahh I see whats going on. 

In order for a component to be saved to a scene, it needs to be registered in the type registry. If a user tried to use this method without registering their types first, then the scene would not be created with that component on it.